hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/OMClientRequest.java

```diff
@@ -140,12 +140,15 @@ public void handleRequestFailure(OzoneManager ozoneManager) {
    *
    * @return the response that will be returned to the client.
    */
-  public abstract OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, TermIndex termIndex);
+  public abstract OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, ExecutionContext context);
 
   /** For testing only. */
   @VisibleForTesting
   public OMClientResponse validateAndUpdateCache(OzoneManager ozoneManager, long transactionLogIndex) {
-    return validateAndUpdateCache(ozoneManager, TransactionInfo.getTermIndex(transactionLogIndex));
+    ExecutionContext context = new ExecutionContext();
+    context.setTermIndex(TransactionInfo.getTermIndex(transactionLogIndex));
+    context.setIndex(transactionLogIndex);
```

Comment:
Please improve safety of `ExecutionContext`:

1. Remove mutators (`set...`)
2. Create private constructors
3. Make member variables `final`
4. Introduce factory methods like:

```java
  public static ExecutionContext of(long index)
  public static ExecutionContext of(TermIndex termIndex)
```

This reduces steps required in code that creates instances, and ensures only valid `ExecutionContext` can be created:

```suggestion
    return validateAndUpdateCache(ozoneManager, ExecutionContext.forIndex(transactionLogIndex));
```

If you plan to add many more members, then add builder instead of factory.

Reply:
@adoroszlai  updated ...

Reply:
Thanks @sumitagrawl for updating the patch.

Original suggestion simplifies callers and also ensures `index` and `termIndex.getIndex()` are consistent:

```java
  public static ExecutionContext of(long index)
  public static ExecutionContext of(TermIndex termIndex)
```

Reply:
@adoroszlai 
This will be different in next PR, so created having together. Only this is to have most files impact to common place in this PR. So above can not be followed.

index: OM managed index -- it will be different of ratis index.
termIndex: Ratis index

Reply:
Will you have non-Ratis `index` and Ratis `TermIndex` with different values at the same time?

Reply:
yes,
ratis index is used in upgrade prepare and snapshot flow.
Other cases, will be using OM managed index (non-ratis).